pkg/handler/collector/github/github.go

```diff
@@ -0,0 +1,327 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/guacsec/guac/internal/client"
+	"github.com/guacsec/guac/internal/client/githubclient"
+	"github.com/guacsec/guac/pkg/collectsub/datasource"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+)
+
+const (
+	GithubCollector = "GithubCollector"
+)
+
+func defaultAssetSuffixes() []string {
+	return []string{".jsonl"}
+}
+
+type TagOrLatest interface {
+	IsTagOrLatest()
+}
+
+type Tag struct {
+	Value string
+}
+
+type Latest struct{}
+
+func (t *Tag) IsTagOrLatest()    {}
+func (l *Latest) IsTagOrLatest() {}
```

Comment:
hmm, i think prefered golang way maybe is something like:

```
type TagOrLatest struct{
    Latest bool
    Tag string
}
```

Reply:
I am still changing it since it's "idiomatic"